Week3/script.js

```diff
@@ -0,0 +1,277 @@
+  }
+  if (code === 3) {
+    type = vehicles[code - 1];
+  }
+  if (code === 4) {
+    type = vehicles[code - 1];
+  }
+  if (age >= 2) {
+    condition = ' used';
+  }
+  if (age >= 0 && age <= 1) {
+    condition = ' new';
+  }
+
+  console.log('a ' + color + condition + ' ' + type);
+}
+newVehicle('green', 3, 1);
+
+// 10
+let advertisement = "Amazing Joe's Garage, we service ";
+function reAdvertisement(advertisement, vehicles) {
+  for (let i = 0; i < vehicles.length; i++) {
+    // const lastElement = vehicles[vehicles.length - 1];
+    // const beforLastElement = vehicles[vehicles.length - 2];
+    // const currentElement = vehicles[i];
+
+    if (i === vehicles.length - 1) {
+      advertisement += vehicles[i] + '.';
+    } else if (i === vehicles.length - 2) {
+      advertisement += vehicles[i] + ' and ';
+    } else {
+      advertisement += vehicles[i] + ', ';
+    }
+  }
+  return advertisement;
+}
+console.log(reAdvertisement(advertisement, vehicles));
+
+// 11
+
+// If you add more vehicle to array, you will be add to advertisement.
+function newFunction() {
+  vehicles.push('truck');
+  return console.log(reAdvertisement(advertisement, vehicles));
+}
+newFunction();
+
+// 12
+
+let newObject = {};
+
+// 13
+
+let hackYourFuture = new Object();
+hackYourFuture.teacher1 = {};
+hackYourFuture.teacher2 = {};
+hackYourFuture.teacher3 = {};
+hackYourFuture.teacher4 = {};
+hackYourFuture.teacher5 = {};
+
+hackYourFuture.teacher1.name = ['Philipp'];
+hackYourFuture.teacher2.name = ['Rob'];
+hackYourFuture.teacher3.name = ['Unmesh'];
+hackYourFuture.teacher4.name = ['Bonan'];
+hackYourFuture.teacher5.name = ['Yash'];
+
+console.log(hackYourFuture);
+
+// 14
+
+hackYourFuture.teacher1.language = ['HTML'];
+hackYourFuture.teacher2.language = ['CSS'];
+hackYourFuture.teacher3.language = ['CLI'];
+hackYourFuture.teacher4.language = ['BonGIT-Bashan'];
+hackYourFuture.teacher5.language = ['JS'];
+
+console.log(hackYourFuture);
+
+// 15
+
+const x = [1, 2, 3];
+const y = [1, 2, 3];
+const z = y;
+
+console.log(x == y); // false - reference is not same
+console.log(x === y); // false - reference is not same
+console.log(z == y); // true - objects returns true when they reference the same memory location.
+console.log(z == x); // false - reference is not same
+
+// 16
+
+const o1 = { foo: 'bar' };
+const o2 = { foo: 'bar' };
+const o3 = o2;
+console.log(o2.foo);
+console.log(o3);
+
+console.log(o1 == o2); // false
+console.log(o2 == o3); // true
+console.log(o3 == o1); // false
+console.log(o2 == o1); // false
+
+// when we change o1 whit o3
+/*
+
+const o3 = { foo: "bar" };
+const o2 = { foo: "bar" };
+const o1 = o2;
+
+*/
+
+console.log(o1 == o2); // true
+console.log(o2 == o3); // false
+console.log(o3 == o1); // false
+console.log(o2 == o1); // false
+
+// when we change o2 whit o3, giving syntax error
+
+// Does the order that you assign (o3 = o2 or o2 = o3) matter?
+
+// sure. when you use o2 = o3, o2 declares twice. giving syntax error.
```

Comment:
Ah could you show this in code? I understand that you explained it in comments, but could you explain it using code?